src/nunit.analyzers/Helpers/AssertHelper.cs

```diff
@@ -46,7 +46,7 @@ public static ITypeSymbol UnwrapActualType(ITypeSymbol actualType)
             if (actualType is INamedTypeSymbol namedType && namedType.DelegateInvokeMethod != null)
                 actualType = namedType.DelegateInvokeMethod.ReturnType;
 
-            if (actualType.IsAwaitable(out var awaitReturnType))
+            if (actualType.IsAwaitable(out var awaitReturnType) && awaitReturnType.SpecialType != SpecialType.System_Void)
                 actualType = awaitReturnType;
 
             return actualType;
```

Comment:
```suggestion
            if (actualType is INamedTypeSymbol namedType && namedType.DelegateInvokeMethod != null)
            {
                actualType = namedType.DelegateInvokeMethod.ReturnType;

	            if (actualType.IsAwaitable(out var awaitReturnType))
	                actualType = awaitReturnType;
            }

            return actualType;
```


Seems like it should unwrap tasks only for delegates.